src/core/mac/link_raw.cpp

```diff
@@ -64,6 +64,21 @@ LinkRaw::LinkRaw(Instance &aInstance)
 {
 }
 
+void LinkRaw::Reset(void)
```

Comment:
Could we call the constructor to reset the class to avoid manually initialize all variables . Otherwise, when we add a new variable to this class, we have to initialize the variable in two functions.
```
void LinkRaw::Reset(Instance &aInstance)
{
    new (this) LinkRaw(aInstance);
}
```

Reply:
Some variables will be intentionally preserved.

In fact `mReceiveDoneCallback` should be preserved, but current codes use it as an indicator of whether the raw link layer is enabled, so I reset it as a workaround.

Reply:
Updated. We can explicitly preserve variables when needed.